config/initializers/elasticsearch.rb

```diff
@@ -1,5 +1,25 @@
 module Elasticsearch; end
 
-es_config = Rails.application.config.elasticsearch_config
+Rails.application.config.elasticsearch = (YAML.load_file("#{Rails.root}/config/elasticsearch.yml") || {})[Rails.env]
 
-Elasticsearch::INDEX_NAME = es_config && es_config['index_name'].present? ? es_config['index_name'].freeze : "#{Rails.env}:jobs".freeze
+config = Rails.application.config.elasticsearch
+
+Elasticsearch::INDEX_NAME = config && config['index_name'].present? ? config['index_name'].freeze : "#{Rails.env}:jobs".freeze
+
+if Rails.env.production?
+  Rails.application.config.elasticsearch_client = Elasticsearch::Client.new(
+    host: [{
+      url: config['url'],
+      user: config['username'],
+      password: config['password']
+    }]
+  )
+else
+  Rails.application.config.elasticsearch_client = Elasticsearch::Client.new(
+    host: config['host'],
```

Comment:
I think you're assuming too much about whether I have a username or password in different environments. If there is a username/password, for the given environment, pass it. If not, don't.

Similarly, you're assuming a lot about whether there are multiple hosts in production or development/test. Have the code work the same in both cases unless there is a good reason not to.